app/models/host/managed.rb

```diff
@@ -669,7 +669,7 @@ def set_hostgroup_defaults
 
   def set_compute_attributes
     return unless compute_profile_id && compute_resource_id
-    self.compute_attributes = compute_resource.compute_profile_attributes_for(compute_profile_id)
+    self.compute_attributes = compute_resource.compute_profile_attributes_for(compute_profile_id).merge(self.compute_attributes || {})
```

Comment:
Is merging the right thing to do, rather than checking if there are already attributes (pre-95d7064)?  It's in progress in PR #2802 so I'm unsure if we should dive in with it here.

That said, I'm not familiar enough with 95d7064 to say whether any of the compute_attributes.empty? can be restored without breaking something else.